homeassistant/components/thermostat/zwave.py

```diff
@@ -0,0 +1,123 @@
+    """Setup the ZWave thermostats."""
+    if discovery_info is None or NETWORK is None:
+      return
+
+    node = NETWORK.nodes[discovery_info[ATTR_NODE_ID]]
+    value = node.values[discovery_info[ATTR_VALUE_ID]]
+    value.set_change_verified(False)
+   # if value.command_class == COMMAND_CLASS_THERMOSTAT_MODE:
+    add_devices([ZWaveThermostat(value)])
+
+
+# pylint: disable=too-many-arguments
+class ZWaveThermostat(ZWaveDeviceEntity, ThermostatDevice, Entity):
+    """Represents a HeatControl thermostat."""
+    def __init__(self, value):
+        from openzwave.network import ZWaveNetwork
+        from pydispatch import dispatcher
+        # ZWaveDeviceEntity.__init__(self, value, DOMAIN)
+        ZWaveDeviceEntity.__init__(self, value, DOMAIN)
+        self._current_temperature = 0
+        self._target_temperature = 71
+        self._unit_of_measurement = TEMP_FAHRENHEIT
+        self._away = False
+        self._current_temperature = 90
+        self._node = value.node
+        self._current_operation = "Idle"
+        self._current_operation_state = "Idle"
+
+        # register listener
+        dispatcher.connect(
+            self.value_changed, ZWaveNetwork.SIGNAL_VALUE_CHANGED)
+
+    def value_changed(self, value):
+      """Called when a value has changed on the network."""
+      if self._value.node == value.node:
+        self.update_properties()
+        self.update_ha_state()
+        print(value.data)
+
+    def update_properties(self):
+      for key, value in self._node.get_values_for_command_class(0x43).items(): # set point
+        if int(value.data) != 0:
+          self._target_temperature = value.data
+      for key, value in self._node.get_values_for_command_class(0x40).items(): # Operation
+        self._current_operation = value.data
+      for key, value in self._node.get_values_for_command_class(0x31).items(): # Current Temp
+        if int(value.data) != 0:
+          self._current_temperature = value.data
+      for key, value in self._node.get_values_for_command_class(0x42).items(): # COMMAND_CLASS_THERMOSTAT_OPERATING_STATE
+        self._current_operation_state = value.data
+
+
+
+
+    @property
+    def should_poll(self):
+        """No polling on ZWave"""
+        return False
+    @property
+    def is_fan_on(self):
+      return self._current_operation_state
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return self._unit_of_measurement
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+    @property
+    def operation(self):
+      return self._current_operation
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    @property
+    def is_away_mode_on(self):
+        """Return if away mode is on."""
+        return self._away
```

Comment:
current_temperature, target_temperature do effect the device. operation should effect the device I just haven't figured out how to do that yet, and I left away mode, so that it could be used in automation, but I will remove it (and check to make sure that the others are effecting the device).